packages/plugin-json/src/lib/validation.ts

```diff
@@ -39,14 +39,39 @@ const isNonNullLocalizedField = (field: FieldDefinitionNode) =>
 export const getTooManyFiltersReport = (types: ObjectTypeDefinitionNode[]) =>
   getFieldsReport(
     types.filter(
-      (type) => (type.fields?.filter(filterableField).length ?? 0) > 5
+      (type) => (type.fields?.filter(filterableField)?.length ?? 0) > 5
     ),
     filterableField
   );
 
 const filterableField = (field: FieldDefinitionNode) =>
   hasDirective(field, "amplienceFilterable");
 
+export const getTooManyDeliveryKeysReport = (types: ObjectTypeDefinitionNode[]) =>
+  getFieldsReport(
+    types.filter(
+      (type) => (type.fields?.filter(deliveryKeyField)?.length ?? 0) > 1
```

Comment:
I believe there is 1 `?` too much since the filter function can never return null/undefined.

```suggestion
      (type) => (type.fields?.filter(deliveryKeyField).length ?? 0) > 1
```

Reply:
Removed (and also for `getTooManyFiltersReport`)